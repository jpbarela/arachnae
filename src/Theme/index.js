// @flow
export type ButtonThemeType = {
  primary: string,
};

export type ThemeType = {
  background: string,
  button: ButtonThemeType,
  color: string,
  fontFamily: string,
};

export const defaultTheme: ThemeType = {
  button: {
    primary: "blue",
  },
  color: "black",
  background: "white",
  fontFamily: "Helvetica, sans-serif",
};
