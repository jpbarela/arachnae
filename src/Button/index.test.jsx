// @flow
import * as React from "react";
import { render } from "@testing-library/react";
import { Button } from "./index";
import { ThemeProvider } from "react-jss";
import { defaultTheme } from "../Theme";

describe("Button", () => {
  it("renders with no style", () => {
    const container = render(
      <ThemeProvider theme={defaultTheme}>
        <Button name="Press Me" />
      </ThemeProvider>
    );
    expect(container.baseElement.firstChild).toMatchSnapshot();
  });

  it("renders with link style", () => {
    const container = render(
      <ThemeProvider theme={defaultTheme}>
        <Button name="Click Me" style="link" />
      </ThemeProvider>
    );
    expect(container.baseElement.firstChild).toMatchSnapshot();
  });

  it("renders with button style", () => {
    const container = render(
      <ThemeProvider theme={defaultTheme}>
        <Button name="Click Me" style="button" />
      </ThemeProvider>
    );
    expect(container.baseElement.firstChild).toMatchSnapshot();
  });
});
