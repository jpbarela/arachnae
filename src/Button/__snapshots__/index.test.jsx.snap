// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<div>
  <button
    class="button-0-2-1"
  >
    Press Me
  </button>
</div>
`;
